datadog/dogshell/timeboard.py

```diff
@@ -174,6 +175,9 @@ def _write_dash_to_file(cls, dash_id, filename, timeout, format='raw', string_id
             if string_ids:
                 dash_obj["id"] = str(dash_obj["id"])
 
+            if not dash_obj.get('template_variables'):
+                dash_obj.pop('template_variables', None)
```

Comment:
What's this for ?

Reply:
So, the way dogshell works is that you tell it to create a new file. Dogshell then calls the API and places the response inside the file on disk. 

You make whatever changes to the file you want, then tell dogshell to update the Datadog timeboard based on the contents of that file. The API read seems to return` template_variables: null` if none are defined. But when you make the update request, its not allowed to be None, the API will return an error. So the easiest solution, IMO is just not to save it to the file if its empty.